### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Dekim12**

### CODE REVIEW

File: src/components/CameraPage/CameraPage.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    const CameraConstants = RNCamera.Constants
+
+    return (
+      <View style={styles.container}>
+        { image ? 
+            (<FastImage 
+              source={{ uri: image.uri, priority: FastImage.priority.high, }} 
+              style={styles.photoStyle} 
+              resizeMode='cover' 
+            />) :
```
Comment (LenaZhuk) [inline src/components/CameraPage/CameraPage.js]:
```
{image ? (
  <Comp />
) : (
  <Comp />
)}
```

File: src/components/ControlBar/ControlBar.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  toggleFlashMode, 
+  isCameraReady, 
+  isImage, 
+  toggleType, 
+}: Props) => (
+  <View style={styles.container}>
+    <TouchableButton style={styles.btnIcons} onPress={toggleFlashMode}>
+      <Icon name='bolt' color={flashMode ? '#3EE7AD' : '#e6e5e6'}/>
+    </TouchableButton>
+    <TouchableButton style={styles.cameraBtn} onPress={isCameraReady ? takePicture : null}>
```
Comment (LenaZhuk) [inline src/components/ControlBar/ControlBar.js]: Maybe it's better just to disable button if a camera isn't ready?

File: src/components/ControlBar/ControlBar.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      { isImage ? (<Icon name='save' size={43}/>) :       
+          isCameraReady ?
```
Comment (LenaZhuk) [inline src/components/ControlBar/ControlBar.js]: Ternary operators should not be nested

File: android/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -2,6 +2,8 @@
   package="com.photoproject">
 
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.CAMERA" />
+    <uses-permission android:name="android.permission.RECORD_AUDIO"/>
```
Comment (LenaZhuk) [inline android/app/src/main/AndroidManifest.xml]: You don't need this permission for photos

File: ios/PhotoProject/Info.plist [hunk 5]
```diff
@@ -71,6 +71,9 @@
 		<string>Octicons.ttf</string>
 		<string>SimpleLineIcons.ttf</string>
 		<string>Zocial.ttf</string>
+		<string>Surveyor Display Bold.ttf</string>
+		<string>Surveyor Display Bold Regular.ttf</string>
+		<string>Edo.ttf</string>
```
Comment (LenaZhuk) [inline ios/PhotoProject/Info.plist:74]: I noticed that you used only `Edo` font

Comment (Dekim12) [reply ios/PhotoProject/Info.plist:74]: fixed

File: android/app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (Dekim12) [reply android/app/src/main/AndroidManifest.xml]: fixed

File: src/components/ControlBar/ControlBar.js (hunk 2 above)
Comment (Dekim12) [reply src/components/ControlBar/ControlBar.js]: fixed

File: src/components/CameraPage/CameraPage.js (hunk 1 above)
Comment (Dekim12) [reply src/components/CameraPage/CameraPage.js]: fixed

File: src/components/ControlBar/ControlBar.js (hunk 3 above)
Comment (Dekim12) [reply src/components/ControlBar/ControlBar.js]: I need it but I wrote it more readable)

File: src/components/ControlBar/ControlBar.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        </TouchableButton>
+        {isCameraReady ? (
+          <TouchableButton
+            style={[styles.cameraBtn, currentCameraBtnStyle]}
+            onPress={takePicture}
+          >
+            {isImage ? (
+              <Icon name='save' size={43} />
+            ) : (
+              <Icon name='camera' size={43} />
```
Comment (LenaZhuk) [inline src/components/ControlBar/ControlBar.js]: Only `name` is different

File: src/components/CameraPage/CameraPage.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+          <RNCamera
+            ref={(ref) => {
+              this.camera = ref
+            }}
+            style={styles.preview}
+            ratio='16:9'
+            captureAudio={false}
+            onCameraReady={this.handleCameraReady}
+            flashMode={
+              flashMode
```
Comment (LenaZhuk) [inline src/components/CameraPage/CameraPage.js]: `isFlashEnabled`
`isFlashOn`

File: src/components/MainPage/MainPage.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+type Props = {
+  toggleCamera: () => void
+}
+
+const MainPage = ({ toggleCamera, }: Props) => (
+      <View style={styles.container}>
+        <Text style={styles.headline}>Photo List</Text> 
+        <PhotoList />   
+        <TouchableButton onPress={toggleCamera} style={styles.makePhotoBtn}>
+        <Text style={styles.btnText}>MAKE A PHOTO</Text>
```
Comment (LenaZhuk) [inline src/components/MainPage/MainPage.js]: Formatting missed

File: src/components/TouchableButton/TouchableButton.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+// @flow
+
+import React from 'react'
+import { TouchableOpacity, } from 'react-native'
+
+import type { Node, } from 'react'
```
Comment (LenaZhuk) [inline src/components/TouchableButton/TouchableButton.js]: `import React, { type Node } from 'react';`

File: src/components/Icon/Icon.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  name: FontAwesome5Glyphs,
+  size: number,
+  color: string,
+  style: ViewStyleProp
+}
+
+const Icon = ({ name, size = 33, color = '#e6e5e6', style, }: Props) => {
+  const IconComponent = Icons
+
+  return <IconComponent name={name} size={size} color={color} style={style} />
```
Comment (LenaZhuk) [inline src/components/Icon/Icon.js:18]: Why not `<Icons ... />`?

Comment (Dekim12) [reply src/components/Icon/Icon.js:18]: Because this component represents only one icon.

File: src/components/ControlBar/ControlBar.js (hunk 6 above)
Comment (Dekim12) [reply src/components/ControlBar/ControlBar.js]: fixed

File: src/components/CameraPage/CameraPage.js (hunk 7 above)
Comment (Dekim12) [reply src/components/CameraPage/CameraPage.js]: fixed

File: src/components/MainPage/MainPage.js (hunk 8 above)
Comment (Dekim12) [reply src/components/MainPage/MainPage.js]: fixed

File: src/components/TouchableButton/TouchableButton.js (hunk 9 above)
Comment (Dekim12) [reply src/components/TouchableButton/TouchableButton.js]: fixed

File: android/app/src/main/java/com/photoproject/MainApplication.java [hunk 11]
```diff
@@ -26,14 +30,9 @@ public boolean getUseDeveloperSupport() {
...
     protected List<ReactPackage> getPackages() {
-      return Arrays.<ReactPackage>asList(
-          new MainReactPackage(),
-            new VectorIconsPackage(),
-            new SvgPackage(),
-            new RNGestureHandlerPackage(),
-            new FastImageViewPackage(),
-            new RNDeviceInfo()
-      );
+      return Arrays.<ReactPackage>asList(new MainReactPackage(), new OrientationPackage(), new RNCameraPackage(),
```
Comment (LenaZhuk) [inline android/app/src/main/java/com/photoproject/MainApplication.java:33]: Formatting

File: src/components/TouchableButton/TouchableButton.js (hunk 9 above)
Comment (LenaZhuk) [reply src/components/TouchableButton/TouchableButton.js]: @Dekim12 please do the same with all imports

File: src/components/CameraPage/CameraPage.js [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+  closeCamera = async (): Promise<void> => {
+    const { toggleCamera, } = this.props
+
+    await this.removePhotoFromCache()
+    toggleCamera()
+  }
+
+  render() {
+    const { isFlashEnabled, image, isCameraReady, isBackType, } = this.state
+    const CameraConstants = RNCamera.Constants
```
Comment (LenaZhuk) [inline src/components/CameraPage/CameraPage.js:113]: Maybe better to use destructuring?

**Pull request closed without merging**